pkg/stanza/fileconsumer/file.go

```diff
@@ -74,13 +66,23 @@ func (m *Manager) Start(persister operator.Persister) error {
 }
 
 func (m *Manager) closePreviousFiles() {
-	if len(m.knownFiles) > 4*m.movingAverageMatches {
-		m.knownFiles = m.knownFiles[m.movingAverageMatches:]
+	// rotate the filesets at end of every consume() call
+
+	// m.knownFiles[0] wil be removed
+	// m.knownFiles[1] -> m.knownFiles[0]
+	// m.previousPollFiles -> m.knownFiles[1]
+
+	for _, r := range m.previousPollFiles.Reset() {
+		m.knownFiles[1].Add(r.Close())
 	}
-	for _, r := range m.previousPollFiles {
-		m.knownFiles = append(m.knownFiles, r.Close())
+}
+
+func (m *Manager) rotateFilesets() {
+	for i := 1; i < len(m.knownFiles); i++ {
+		m.knownFiles[i-1].Clear()
+		m.knownFiles[i-1] = m.knownFiles[i].Copy()
+		m.knownFiles[i].Clear()
```

Comment:
This is still focused on moving files between file sets. Can't we just rotate the filesets themselves?

```suggestion
func (m *Manager) rotateFilesets() {
	for i := 0; i+1 < len(m.knownFiles); i++ {
		m.knownFiles[i] = m.knownFiles[i+1]
	}
	m.knownFiles[len(m.knownFiles)-1] = fileset.New...
```

Reply:
Sorry about it, I am still working on this.
Will take care of this before final review.

Reply:
No worries, just thought you were probably waiting on me. Thanks for all your work on this. We're getting there.